### PR TITLE
control flow graphs: introduce 'join' points for easy analyses based …

### PR DESCRIPTION
**Pull request by Araq**
…on abstract interpretation

### CODE REVIEW

File: compiler/sempass2.nim [hunk 1]
```diff
@@ -1013,7 +1013,7 @@ proc trackProc*(g: ModuleGraph; s: PSym, body: PNode) =
       "declared lock level is $1, but real lock level is $2" %
         [$s.typ.lockLevel, $t.maxLockLevel])
   when defined(useDfa):
-    if s.kind == skFunc:
+    if s.name.s == "testp":
```
Comment (Clyybber) [inline compiler/sempass2.nim]: I assume this is a leftover from testing?

Comment (Araq) [reply compiler/sempass2.nim]: ``useDfa`` is usually not defined anyway.

**Comment by Araq** [comment]
@cooldome can you test this branch with your destructor heavy code? I rewrote the lastReadOf analysis. It's "correct by construction" but hey, you never know...

**Comment by Araq** [comment]
Well, we can always revert it later, merging.

**Pull request merged**